ops/manifests/manipulations.py

```diff
@@ -167,7 +168,13 @@ def __call__(self, obj: AnyResource):
                 "juju.io/manifest": self.manifests.name,
                 "juju.io/manifest-version": f"{self.manifests.name}-{version}",
             }
-            obj.metadata.labels = obj.metadata.labels or {}  # ensure object has labels
+            if isinstance(obj, (GenericGlobalResource, GenericNamespacedResource)):
+                # Custom resources in lightkube are built differently
+                # from standard model resources
+                obj["metadata"]["labels"] = obj.metadata.labels or {}
```

Comment:
how does `obj["metadata"]["labels"]` work differently from `obj.metadata.labels`

Reply:
So, the `GenericGlobalResource` and `GenericNamespacedResource` are base classes for the class used to create custom object types in lightkube. They both are types of [`Generic`](https://github.com/gtsystem/lightkube/blob/master/lightkube/generic_resource.py#L37) which are fundamentally dicts with super powers.

this is different from modeled objects like say `DaemonSet` which is generated from the upstream definitions of that object and inherits from [`DataclassDictMixin`](https://github.com/gtsystem/lightkube-models/blob/master/lightkube-generate/templates/models.tmpl#L15).  That baseclass doesn't matter much, but the `Generic` object is meant to function like a real resource type -- but it's not a first class object. 

It does a nice job for 'reading' these resources if they're in yaml -- but it's unproven yet if this PR will actually work for updating those custom objects.  I'll let you know if there's more to do here or if i'm way off base -- but just starting this PR to hold my work until I've proven I can use lightkube to patch CRD objects

These were the starting points I had to take just to get the ops-lib-manifest to read the yaml manifests out of the charm -- much less actually manipulate then patch into the cluster

Reply:
alright that makes sense. So the the generic resources get created as dicts (which wont have object attributes like regular upstream resources would). Thanks!

Reply:
![image](https://user-images.githubusercontent.com/10090033/202297114-8e138289-1ad6-47d2-b483-e4c5c9ea7f34.png)


oh yeah.  those labels get applied!!